File: code/server/src/Core/Entities/AuditLog.cs
using IMT_Reservas.Server.Core.Abstraction;

namespace IMT_Reservas.Server.Core.Entities;

public class AuditLog : Entity
{
    public string AdminCarnet { get; set; } = string.Empty;
    public string AdminNombre { get; set; } = string.Empty;
    public string Accion { get; set; } = string.Empty;
    public string Entidad { get; set; } = string.Empty;
    public string? EntidadId { get; set; }
    public string? Detalle { get; set; }
    public DateTime Timestamp { get; set; } = DateTime.UtcNow;
}

File: code/client/src/app/entities/admin/model/audit-log.ts
export interface AuditLogDto {
  Id?: number;
  AdminCarnet?: string;
  AdminNombre?: string;
  Accion?: string;
  Entidad?: string;
  EntidadId?: string;
  EntidadNombre?: string;
  Detalle?: string;
  Timestamp?: Date;
}
